File: Stats.CPP/Wizard_MLReg.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "Wizard_MLReg.h"
#include "MLRegWizard.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "Basic_Form"
#pragma resource "*.dfm"
TfrmShowMLRWizard *frmShowMLRWizard;
//---------------------------------------------------------------------------
__fastcall TfrmShowMLRWizard::TfrmShowMLRWizard(TComponent* Owner)
	: TfrmBasic(Owner)
{
	Memo1->Lines->Clear();
	Memo1->Lines->Add("By using TMtxMulLinReg component you can easiliy perform"
			" multiple linear regression. Just add a matrix of independent"
			" variables, vector of dependent variable and (optionally)"
			" weights and let Statistics regression routines do the rest.");

	TfrmMulLinRegWiz* frmMulLinRegWiz = new TfrmMulLinRegWiz(this);
	frmMulLinRegWiz->Align = alClient;
	frmMulLinRegWiz->Parent = this;
	frmMulLinRegWiz->BorderStyle = bsNone;
	frmMulLinRegWiz->btnCancel->Visible = false; // in this case hide Cancel button
	// load prepared data

	frmMulLinRegWiz->MtxMulLinReg->Y->LoadFromFile(CodePath() + "MulLinReg_Y.vec");
	frmMulLinRegWiz->MtxMulLinReg->A->LoadFromFile(CodePath() + "MulLinReg_A.mtx");

	frmMulLinRegWiz->Show();
}
//---------------------------------------------------------------------------

File: Stats.Delphi/Wizard_MLReg.pas
unit Wizard_MLReg;

interface

uses
  Windows, Messages, SysUtils, Classes, Graphics, Controls, Forms, Dialogs,
  Basic_Form, StdCtrls, ExtCtrls;
                              
type
  TfrmShowMLRWizard = class(TfrmBasic)
    procedure FormCreate(Sender: TObject);
  private
    { Private declarations }
  public
    { Public declarations }
  end;

var
  frmShowMLRWizard: TfrmShowMLRWizard;

implementation

uses MLRegWizard;  { <--- Custom wizard, you can chage it at will <g> }

{$R *.DFM}

procedure TfrmShowMLRWizard.FormCreate(Sender: TObject);
begin
  inherited;
  With Memo1.Lines do
  begin
    Clear;
    Add('By using TMtxMulLinReg component you can easiliy perform'+
        ' multiple linear regression. Just add a matrix of independent'+
        ' variables, vector of dependent variable and (optionally)'+
        ' weights and let Statistics regression routines do the rest.');
  end;

  With TfrmMulLinRegWiz.Create(Self) do
  begin
    Align := alClient;
    Parent := Self;
    BorderStyle := bsNone;
    BtnCancel.Visible := false; { in this case hide Cancel button }
    { load prepared data }
    try
      MtxMulLinReg.Y.LoadFromFile('MulLinReg_Y.vec');
      MtxMulLinReg.A.LoadFromFile('MulLinReg_A.mtx');
    except
    end;
    Show();
  end;
end;

initialization
  RegisterClass(TfrmShowMLRWizard);

end.
